### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by byteZorvin**
- Adds unit tests with mock testing method

### CODE REVIEW

File: src/bridge/tests/token_actions_test.cairo [hunk 1]
```diff
@@ -217,3 +333,113 @@ fn consume_message_zero_recipient() {
...
+    mock.send_deploy_message(usdc_address);
+}
+
+#[test]
+fn send_deposit_message_ok() {
+    let mut mock = mock_state_testing();
+    let usdc_address = USDC_MOCK_ADDRESS();
+
+    // Deploy messaging mock with 5 days cancellation delay
+    let messaging_mock_class_hash = snf::declare("messaging_mock").unwrap();
```
Comment (apoorvsadana) [inline src/bridge/tests/token_actions_test.cairo:370]: can abstract these parts into a common setup function

File: src/bridge/tests/token_actions_test.cairo [hunk 2]
```diff
@@ -217,3 +333,113 @@ fn consume_message_zero_recipient() {
...
+    // Deploying with 5 days as the delay time (5 * 86400 = 432000)
+    let (messaging_contract_address, _) = messaging_mock_class_hash
+        .deploy(@array![DELAY_TIME])
+        .unwrap();
+
+    TokenBridge::constructor(ref mock, L3_BRIDGE_ADDRESS(), messaging_contract_address, OWNER());
+
+    let no_message: Span<felt252> = array![].span();
+    mock
```
Comment (apoorvsadana) [inline src/bridge/tests/token_actions_test.cairo:379]: we should check if the message was received

File: tests/token_actions_test.cairo [hunk 3]
```diff
@@ -139,3 +139,21 @@ fn disable_withdrawal_limit_not_owner() {
...
+    let withdrawal_limit = IWithdrawalLimitStatusDispatcher {
+        contract_address: token_bridge.contract_address
+    };
+
+    snf::start_cheat_caller_address(token_bridge.contract_address, OWNER());
+    let usdc_address = USDC_MOCK_ADDRESS();
+    token_bridge_admin.enable_withdrawal_limit(usdc_address);
+    snf::stop_cheat_caller_address(token_bridge.contract_address);
+
+    assert(withdrawal_limit.is_withdrawal_limit_applied(usdc_address), 'Limit not applied');
```
Comment (apoorvsadana) [inline tests/token_actions_test.cairo:158]: let's check before calling `enable_withdrawal_limit` it was false

File: src/bridge/tests/token_actions_test.cairo [hunk 4]
```diff
@@ -217,3 +333,113 @@ fn consume_message_zero_recipient() {
...
+    let mut mock = mock_state_testing();
+    let usdc_address = USDC_MOCK_ADDRESS();
+
+    // Setting the token active
+    let old_settings = mock.token_settings.read(usdc_address);
+    mock
+        .token_settings
+        .write(usdc_address, TokenSettings { token_status: TokenStatus::Active, ..old_settings });
+
+    assert(mock.is_servicing_token(usdc_address) == true, 'Should be servicing');
```
Comment (apoorvsadana) [inline src/bridge/tests/token_actions_test.cairo:443]: let's check before setting the status it was false

File: tests/token_actions_test.cairo (hunk 3 above)
Comment (byteZorvin) [reply tests/token_actions_test.cairo:158]: updated

File: src/bridge/tests/token_actions_test.cairo (hunk 4 above)
Comment (byteZorvin) [reply src/bridge/tests/token_actions_test.cairo:443]: updated

File: src/bridge/tests/token_actions_test.cairo (hunk 2 above)
Comment (byteZorvin) [reply src/bridge/tests/token_actions_test.cairo:379]: updated

File: src/bridge/tests/token_actions_test.cairo [hunk 5]
```diff
@@ -217,3 +333,113 @@ fn consume_message_zero_recipient() {
...
+    let messaging_mock_class_hash = snf::declare("messaging_mock").unwrap();
+    // Deploying with 5 days as the delay time (5 * 86400 = 432000)
+    let (messaging_contract_address, _) = messaging_mock_class_hash
+        .deploy(@array![DELAY_TIME])
+        .unwrap();
+
+    snf::start_cheat_caller_address_global(snf::test_address());
+    TokenBridge::constructor(ref mock, L3_BRIDGE_ADDRESS(), messaging_contract_address, OWNER());
+
+    mock.send_deploy_message(usdc_address);
```
[the remark this answers is not in the document]
Comment (byteZorvin) [reply src/bridge/tests/token_actions_test.cairo:352]: updated

File: src/bridge/tests/token_actions_test.cairo [hunk 6]
```diff
@@ -207,6 +268,61 @@ fn enroll_token_blocked() {
 }
 
 
+#[test]
+fn consume_message_ok() {
```
[the remark this answers is not in the document]
Comment (byteZorvin) [reply src/bridge/tests/token_actions_test.cairo:272]: Create a new file `bridge/tests/messaging_test.cairo`

File: src/bridge/tests/token_actions_test.cairo (hunk 1 above)
Comment (byteZorvin) [reply src/bridge/tests/token_actions_test.cairo:370]: Will address this in the #6

File: tests/withdrawal_limit_bridge_test.cairo [hunk 7]
```diff
@@ -139,3 +139,25 @@ fn disable_withdrawal_limit_not_owner() {
...
+    let usdc_address = USDC_MOCK_ADDRESS();
+    let token_bridge_admin = ITokenBridgeAdminDispatcher {
+        contract_address: token_bridge.contract_address
+    };
+    let withdrawal_limit = IWithdrawalLimitStatusDispatcher {
+        contract_address: token_bridge.contract_address
+    };
+
+    assert(
+        withdrawal_limit.is_withdrawal_limit_applied(usdc_address) == false, 'Limit not applied'
```
Comment (apoorvsadana) [inline tests/withdrawal_limit_bridge_test.cairo:155]:
```suggestion
        withdrawal_limit.is_withdrawal_limit_applied(usdc_address) == false, 'Limit already applied'
```

**Pull request merged**